docs/workflows/build-pages-with-storybook.md

```diff
@@ -64,35 +64,103 @@ In such cases it is natural to use [args composition](../writing-stories/args.md
+If you are using a provider that supplies data via the context, you can wrap your story in a decorator that provides a mocked version of that provider. For example, in the [Screens](https://storybook.js.org/tutorials/intro-to-storybook/react/en/screen/) chapter of the Intro to Storybook tutorial, we mock a Redux provider with mock data.
 
-Additionally, there may be addons that supply such providers and nice APIs to set the data they provide. For instance [`storybook-addon-apollo-client`](https://www.npmjs.com/package/storybook-addon-apollo-client) provides this API:
+### Mocking API Services
+
+Connected applications are everywhere, consuming data either from REST or GraphQL endpoints. If you're working in an application that relies on either of these data providers, you can add Mock Service Worker (MSW) and [Storybook's MSW addon](https://storybook.js.org/addons/msw-storybook-addon) to mock data alongside your app and stories.
+
+[Mock Service Worker](https://mswjs.io/) is an API mocking library. It relies on service workers to capture network requests and provides mocked data in response. The MSW addon adds this functionality into Storybook, allowing you to mock API requests in your stories.
+
+#### Mocking REST requests with MSW addon
+
+The approach documented [above](#args-composition-for-presentational-screens) could be quickly adapted if you're relying on a Restful API. For instance, if you're retrieving the list of documents from a Restful endpoint, you could have an implementation similar to:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/document-screen-fetch.js.mdx',
+    'vue/document-screen-fetch.3.js.mdx',
+    'angular/document-screen-fetch.ts.mdx',
+    'svelte/document-screen-fetch.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+To test your screen with the mocked data, you could write a similar set of stories:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/documentscreen-story-msw-rest-request.js.mdx',
+    'vue/documentscreen-story-msw-rest-request.3.js.mdx',
+    'angular/documentscreen-story-msw-rest-request.ts.mdx',
+    'svelte/documentscreen-story-msw-rest-request.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+💡 <strong>Note:</strong> This example details how you can mock the REST request with fetch. Similar HTTP clients such as <a href="https://axios-http.com/">axios</a> can be used as well.
+</div>
+
+The mocked data (i.e., `TestData`) will be injected via [parameters](./writing-stories/parameters), enabling you to configure it per-story basis.
+
+#### Mocking GraphQL queries with MSW addon
+
+If instead of a Restful endpoint, you're retrieving the data from a [GraphQL](https://www.apollographql.com/docs/react/integrations/integrations/) endpoint. You could have the following implementation for your screen:
```

Comment:
Start this paragraph by introducing the subject again instead of referencing the previous paragraph. That way we can link directly to this paragraph without the reader needing to scroll up for more context.